tests/Fixtures/Admin/CustomQueryAdmin.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sonata\AdminBundle\Tests\Fixtures\Admin;
+
+use Doctrine\Common\Collections\Criteria;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Admin\FieldDescriptionCollection;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+
+final class CustomQueryAdmin extends AbstractAdmin
+{
+    public function createQuery($context = 'list'): ProxyQueryInterface
```

Comment:
`createQuery()` is marked as final, you should use `configureQuery()` instead.

Reply:
Using a Mock could be even better.

```
$query = $this->createMock(ProxyQueryInterface::class);
$query->method('getSortBy')->willReturn(...);
$query->method('getSortOrder')->willReturn(...);

$admin = $this->createMock(AdminInterface::class);
$admin->method('createQuery')->willReturn($query);
```

Then we won't need to create both CustomQueryAdmin and CustomQuery.

Reply:
I've pushed 26517feb commit with mocks, but there is an error:
```php
Error : Call to a member function getValues() on null
tests/Admin/AdminTest.php:1888
```
I cannot mock `AbstractAdmin::getDatagrid()` method, because it calls `AbstractAdmin::buildDatagrid()` method that calls `AbstractAdmin::determineSorting()`. The last one is the method that I would to test.

Reply:
Done (26517feb commit)